tests/general/InvalidLinkBearTest.py

```diff
@@ -1,53 +1,113 @@
-                                        ["http://coala-analyzer.org/\n"],
-                                        ["http://coala-analyzer.org/"],
-                                        ["http://not a link dot com"],
-                                        ["<http://lwn.net/>"],
-                                        ["'https://www.gnome.org/'"],
-                                        ["http://coala-analyzer.org/..."]),
-                                     invalid_files=(
-                                        ["http://coalaisthebest.com"],),
-                                     force_linebreaks=False)
-
-MarkdownLinks = verify_local_bear(InvalidLinkBear,
-                                  valid_files=(
-                                        ["[coala](http://coala-analyzer.org/)"],
-                                        ["https://en.wikipedia.org/wiki/"
-                                         "Hello_(Adele_song)"]),
-                                  invalid_files=(),
-                                  force_linebreaks=False)
-
-SphinxLinks = verify_local_bear(InvalidLinkBear,
-                                valid_files=(
-                                    ["|https://github.com/coala-analyzer/"
-                                     "coala-bears|"],),
-                                invalid_files=(),
-                                force_linebreaks=False)
+from coalib.settings.Section import Section
+
+
+def custom_matcher(request):
+    change_url = False
+    try:
+        status_code = int(request.path_url[-3:])
+        if status_code in range(300, 400):
+            change_url = True
+            url = "some_url"
+        resp = requests.Response()
+        if change_url:
+            resp.url = url
+        resp.status_code = status_code
+        return resp
+    except ValueError:
+        raise requests.exceptions.RequestException("")
+
+
+class InvalidLinkBearTest(unittest.TestCase):
+
+    def setUp(self):
+        self.section = Section("")
+
+    def assertResult(self, valid_file=None, invalid_file=None):
+        with requests_mock.Mocker() as m:
+            InvalidLinkBear.check_prerequisites = lambda *args: True
+            uut = InvalidLinkBear(self.section, Queue())
+            m.add_matcher(custom_matcher)
+            if valid_file:
+                out = uut.execute("valid", valid_file)
+                self.assertEqual(out, [])
+            if invalid_file:
+                out = uut.execute("invalid", invalid_file)
+                self.assertNotEqual(out, [])
+                self.assertNotEqual(out, None)
+
+    def test_run(self):
+        # Link Redirect
+        valid_file = """
+        http://httpbin.org/status/200
+        """.splitlines()
+
+        invalid_file = """
+        http://httpbin.org/status/301
+        http://coala.rtfd.org/302
+        """.splitlines()
+
+        self.assertResult(valid_file=valid_file, invalid_file=invalid_file)
```

Comment:
for future reference you usually want to split this up into several test_ functions